examples/linear_elasticity/linear_elasticity_model.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import numpy as np
+
+from fenics_constitutive import Constraint, IncrSmallStrainModel, strain_from_grad_u
+
+
+class LinearElasticityModel(IncrSmallStrainModel):
+    def __init__(self, parameters: dict[str, float], constraint: Constraint):
+        self._constraint = constraint
+        E = parameters["E"]
+        nu = parameters["nu"]
```

Comment:
Would it make sense (and also work with the below definition of `np.ndarray`s) to allow `parameters: dict[str, float | dolfinx.fem.Constant]` ?
For the standard elasticity implementation via UFL, you can update the material and reassemble the system matrix without re-creating the whole model/problem by changing the value of the constant.

Something like
```python
E = fem.Constant(domain, value)
NU = fem.Constant(domain, value)
problem = LinearElasticityProblem(E, NU)
problem.assemble_system()
u = problem.solve()

# update and solve again
E.value = new_value
NU.value = new_value
problem.assemble_system()
other = problem.solve()
```

If you don't think this would be a relevant use case that's also fine, but I guess you cannot pass a `fem.Constant` as input to the `array`

Reply:
In the use-case that you describe, I would be more in favor of not using this linear elasticity model, since it mainly serves the purpose of showing how to implement a constitutive model, and also to be interchangeable with other constitutive models.

Also you are probably right that putting the `fem.Constant` into the array does not really make sense. Although numpy arrays can have any datatype (at least I think so), we work on the vector of the functionspace, which is alwys some kind of `float`.

Reply:
> In the use-case that you describe, I would be more in favor of not using this linear elasticity model, since it mainly serves the purpose of showing how to implement a constitutive model, and also to be interchangeable with other constitutive models.

Okay, makes sense. I agree.